libraries/AP_OSD/AP_OSD_Screen.cpp

```diff
@@ -1077,6 +1077,16 @@ const AP_Param::GroupInfo AP_OSD_Screen::var_info2[] = {
     // @User: Standard
     AP_GROUPINFO("FONT", 4, AP_OSD_Screen, font_index, 0),
 #endif
+
+#if HAL_WITH_MSP_DISPLAYPORT
+    // @Param: TXT_SCALE
+    // @DisplayName: Scales OSD element positions based on the the overlay text resolution (MSP DisplayPort only)
+    // @Description: Scales OSD element positions based on the the overlay text resolution (MSP DisplayPort only)
+    // @Values: 0:Disable,1:Enabled
+    // @User: Standard
+    AP_GROUPINFO("TXT_SCALE", 5, AP_OSD_Screen, txt_scale, 0),
```

Comment:
aren't we better off with a table of params for hires, and we loop over the table calling set_default() to shift the positions of the default elements? That would leave MissionPlanner correct and all the parameters would reflect the true positions

Reply:
I thought of that but would be expecting some gnashing of teeth about the flash cost, so I didn't suggest it. There are a **lot** of OSD parameters.

Reply:
Note that things are already not correct with fixed positions since PAL and NTSC have different number of lines.

Reply:
the positions are correct with either osd type, just may  not be visible if setuo for pal on a  ntsc vtx....and MP shows both pal and ntsc layouts correctly

Reply:
> how much more, I do not know but there are a lot of OSD elements.

we would only need elements in the table which are enabled by default, and only need the x/y parameters. We could also use references instead of names in the table. I would not be surprised if it came out to less flash than this PR. This PR costs 1k of flash.